.github/workflows/clang-format-check.yml

```diff
@@ -0,0 +1,24 @@
+name: clang-format
+on: [push, pull_request]
+concurrency:
+  group: ${{ github.workflow }}-${{ github.job }}-${{ github.ref }}
+  cancel-in-progress: true
+jobs:
+  formatting-check:
+    name: Formatting check
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        path:
+          - 'include'
+          - 'test'
+          - 'benchmark'
+    steps:
+    - uses: actions/checkout@v3
+    - name: Run clang-format style check for C/C++/Protobuf programs.
+      uses: jidicula/clang-format-action@v4.10.1
+      with:
+        clang-format-version: '15'
+        # exclude-regex: '(tests\/catch.hpp|benchmarks\/catch.hpp)'
+        check-path: ${{ matrix.path }}
```

Comment:
I think we should mimic (if not copy) the `.pre-commit-config.yaml` and `.github/workflow/static-analysis.yml` from Xtensor repo. It will be easier to maintain.

Reply:
Agreed @AntoinePrv, I didn't see that until now, I'll update just now.

Reply:
@AntoinePrv I think I've done this now (and I recommitted so that all the changes from `pre-commit` were enacted).